mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,530 @@
+    for (key, value) in maker_orders.iter() {
+        let key_trade_pair = TradingPair::new(value.base.clone(), value.rel.clone());
+        match cfg.get(&key_trade_pair.as_combination()) {
+            Some(coin_cfg) => {
+                // res will be used later for reporting error to the users, also usefullt o be coupled with a telegram service to send notification to the user
+                let _res = update_single_order(
+                    coin_cfg.clone(),
+                    *key,
+                    value.clone(),
+                    key_trade_pair.as_combination(),
+                    ctx,
+                )
+                .await;
+                memoization_pair_registry.insert(key_trade_pair.as_combination());
+            },
+            _ => continue,
+        }
+        println!("{}", key);
+    }
+
+    // Now iterate over the registry and for every pairs that are not hit let's create an order
+    for (trading_pair, cur_cfg) in cfg.iter() {
+        match memoization_pair_registry.get(trading_pair) {
+            Some(_) => continue,
+            None => {
+                // res will be used later for reporting error to the users, also usefullt o be coupled with a telegram service to send notification to the user
+                let _res = create_single_order(cur_cfg.clone(), trading_pair.clone(), ctx).await;
+            },
+        };
+    }
+}
+
+pub async fn lp_bot_loop(ctx: MmArc) {
+    info!("lp_bot_loop successfully started");
+    loop {
+        // todo: this log should probably in debug
+        info!("tick lp_bot_loop");
+        if ctx.is_stopping() {
+            // todo: can we cancel all the pending orders when the ctx is stopping or call tear_down ?
+            break;
+        }
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+        let mut states = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        if *states == TradingBotState::Stopping {
+            *states = TradingBotState::Stopped;
+            // todo: verify if there is a possible deadlock here if i use states inside tear_down_bot
+            tear_down_bot(ctx).await;
+            break;
+        }
+        drop(states);
+        process_bot_logic(&ctx).await;
+        Timer::sleep(30.0).await;
+    }
+    info!("lp_bot_loop successfully stopped");
+}
+
+pub async fn process_price_request() -> Result<(StatusCode, String, HeaderMap), MmError<PriceServiceRequestError>> {
+    info!("Fetching price from: {}", KMD_PRICE_ENDPOINT);
+    let (status, headers, body) = slurp_url(KMD_PRICE_ENDPOINT).await?;
+    Ok((status, std::str::from_utf8(&body)?.trim().into(), headers))
+}
+
+async fn fetch_price_tickers(ctx: &MmArc) {
+    let (status_code, body, _) = match process_price_request().await {
+        Ok(x) => x,
+        Err(_) => return,
+    };
+    if status_code == StatusCode::OK {
```

Comment:
Please move this inside `process_price_request`. The `status_code != StatusCode::OK` and parse errors should lead to the error of `process_price_request`.